--- lightdm/configure.ac	2011-06-15 05:25:43 +0000
+++ lightdm/configure.ac	2011-06-17 07:14:07 +0000
@@ -36,6 +36,10 @@ PKG_CHECK_MODULES(GLIB, [
     glib-2.0
 ])
 
+PKG_CHECK_MODULES(GOBJECT, [
+    gobject-2.0
+])
+
 PKG_CHECK_MODULES(XCB, [
     xcb
 ])    

--- lightdm/tests/src/Makefile.am	2011-06-15 05:25:43 +0000
+++ lightdm/tests/src/Makefile.am	2011-06-17 07:14:07 +0000
@@ -20,11 +20,13 @@ test_gobject_greeter_CFLAGS = \
 	-I$(top_srcdir)/liblightdm-gobject \
 	$(WARN_CFLAGS) \
 	$(GLIB_CFLAGS) \
+	$(GOBJECT_CFLAGS) \
 	$(XCB_CFLAGS)
 test_gobject_greeter_LDADD = \
 	-L$(top_builddir)/liblightdm-gobject \
 	-llightdm-gobject-0 \
 	$(GLIB_LIBS) \
+	$(GOBJECT_LIBS) \
 	$(XCB_LIBS)
 
 test_session_SOURCES = test-session.c status.c status.h

