Author: Yves-Alexis Perez <corsac@debian.org>
Author: Lionel Le Folgoc <mrpouit@gmail.com>
Bug: https://bugs.launchpad.net/ubuntu/+source/lightdm/+bug/798277
Description: Fix default PATH environment variable
Index: lightdm/src/seat.c
===================================================================
--- lightdm.orig/src/seat.c	2013-08-24 19:45:09.358830075 +1000
+++ lightdm/src/seat.c	2013-08-24 19:45:09.354830059 +1000
@@ -241,7 +241,10 @@
         process_set_env (script, "PATH", g_getenv ("PATH"));
     }
     else
-        process_set_env (script, "PATH", "/usr/local/bin:/usr/bin:/bin");
+        if (user_get_uid (user) == 0)
+          process_set_env (script, "PATH", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin");
+        else
+          process_set_env (script, "PATH", "/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games");
 
     if (user)
     {
Index: lightdm/src/session-child.c
===================================================================
--- lightdm.orig/src/session-child.c	2013-08-24 19:45:09.358830075 +1000
+++ lightdm/src/session-child.c	2013-08-24 19:45:09.354830059 +1000
@@ -355,7 +355,10 @@
         else
         {
             /* Set POSIX variables */
-            pam_putenv (pam_handle, "PATH=/usr/local/bin:/usr/bin:/bin");
+            if (user_get_uid (user) == 0)
+              pam_putenv (pam_handle, "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin");
+            else
+              pam_putenv (pam_handle, "PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games");
             pam_putenv (pam_handle, g_strdup_printf ("USER=%s", username));
             pam_putenv (pam_handle, g_strdup_printf ("LOGNAME=%s", username));
             pam_putenv (pam_handle, g_strdup_printf ("HOME=%s", user_get_home_directory (user)));
