Author: Yves-Alexis Perez <corsac@debian.org>
Author: Lionel Le Folgoc <mrpouit@gmail.com>
Bug: https://bugs.launchpad.net/ubuntu/+source/lightdm/+bug/798277
Description: Fix default PATH environment variable
Index: lightdm-0.9.5/src/session.c
===================================================================
--- lightdm-0.9.5.orig/src/session.c	2011-09-07 07:32:43.000000000 +0200
+++ lightdm-0.9.5/src/session.c	2011-09-07 09:21:36.280133903 +0200
@@ -199,7 +199,10 @@
     pam_session_open (session->priv->authentication);
 
     user = pam_session_get_user (session->priv->authentication);
-    process_set_env (PROCESS (session), "PATH", "/usr/local/bin:/usr/bin:/bin");
+    if (user_get_uid (user) == 0)
+      process_set_env (PROCESS (session), "PATH", "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin");
+    else
+      process_set_env (PROCESS (session), "PATH", "/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games");
     process_set_env (PROCESS (session), "USER", user_get_name (user));
     process_set_env (PROCESS (session), "USERNAME", user_get_name (user)); // FIXME: Is this required?
     process_set_env (PROCESS (session), "HOME", user_get_home_directory (user));
