=== modified file 'data/lightdm.conf'
Index: lightdm-0.3.0/data/lightdm.conf
===================================================================
--- lightdm-0.3.0.orig/data/lightdm.conf	2011-04-16 10:16:26.000000000 +0200
+++ lightdm-0.3.0/data/lightdm.conf	2011-04-18 14:44:13.498641953 +0200
@@ -9,9 +9,9 @@
 #
 [LightDM]
 #xserver=/usr/bin/X
-#authorization-directory=/var/cache/lightdm/authority
+authorization-directory=/var/lib/lightdm/authority
 #log-directory=/var/log/lightdm
-#session-wrapper=/etc/X11/Xsession
+session-wrapper=/etc/X11/Xsession
 displays=default-display
 
 #
@@ -31,7 +31,7 @@
 # key = Authentication key to use
 #
 [default-display]
-#vt=
+vt=7
 #layout=us
 #greeter-theme=example-gtk-gnome
 #greeter-user=lightdm
@@ -54,7 +54,7 @@
 # hidden-shells = Shells that indicate a user cannot login
 #
 [UserManager]
-#load-users=true
+load-users=false
 #minimum-uid=500
 #hidden-users=nobody nobody4 noaccess
 #hidden-shells=/bin/false /usr/sbin/nologin
@@ -70,6 +70,6 @@
 # it can be a word and the first 7 characters are used as the key.
 #
 [xdmcp]
-#enabled=true
+enabled=false
 #port=177
 #key=0x0123456789ABCD
