Index: lightdm-0.9.4/greeters/gtk/lightdm-gtk-greeter.conf
===================================================================
--- lightdm-0.9.4.orig/greeters/gtk/lightdm-gtk-greeter.conf	2011-07-22 05:24:36.000000000 +0200
+++ lightdm-0.9.4/greeters/gtk/lightdm-gtk-greeter.conf	2011-09-07 09:21:05.291942506 +0200
@@ -8,10 +8,10 @@
 # xft-rgba = Type of subpixel antialiasing (none, rgb, bgr, vrgb or vbgr)
 #
 [greeter]
-#background=
-#theme-name=
+background=/usr/share/images/desktop-base/desktop-background
+theme-name=Adwaita
 #font-name=
-#xft-antialias=
+xft-antialias=true
 #xft-dpi=
-#xft-hintstyle=
-#xft-rgba=
+xft-hintstyle=hintfull
+xft-rgba=rgb
