Index: lightdm-1.0.0/greeters/gtk/lightdm-gtk-greeter.conf
===================================================================
--- lightdm-1.0.0.orig/greeters/gtk/lightdm-gtk-greeter.conf	2011-09-20 03:45:23.000000000 +0200
+++ lightdm-1.0.0/greeters/gtk/lightdm-gtk-greeter.conf	2011-09-28 13:18:07.865571415 +0200
@@ -8,11 +8,11 @@
 # xft-rgba = Type of subpixel antialiasing (none, rgb, bgr, vrgb or vbgr)
 #
 [greeter]
-#background=
+background=/usr/share/images/desktop-base/desktop-background
 #show-language-selector=false
-#theme-name=
+theme-name=Adwaita
 #font-name=
-#xft-antialias=
+xft-antialias=true
 #xft-dpi=
-#xft-hintstyle=
-#xft-rgba=
+xft-hintstyle=hintfull
+xft-rgba=rgb
