Author: Yves-Alexis Perez <corsac@debian.org>
Description: don't add PKGLIBEXEC_DIR to the user PATH
--- a/src/session-child.c
+++ b/src/session-child.c
@@ -479,11 +479,6 @@ session_child_run (int argc, char **argv
         g_free (value);
     }
 
-    /* Put our tools directory in the path as a hack so we can use the legacy gdmflexiserver interface */
-    path = pam_getenv (pam_handle, "PATH");
-    if (path)
-        pam_putenv (pam_handle, g_strdup_printf ("PATH=%s:%s", PKGLIBEXEC_DIR, path));
-
     /* Catch terminate signal and pass it to the child */
     signal (SIGTERM, signal_cb);
 
