Index: lightdm-0.4.0/src/display-manager.c
===================================================================
--- lightdm-0.4.0.orig/src/display-manager.c	2011-06-17 04:04:18.000000000 +0200
+++ lightdm-0.4.0/src/display-manager.c	2011-06-17 08:38:37.778236133 +0200
@@ -137,14 +137,6 @@
     }
 
     authorization = xserver_get_authorization (display_get_xserver (display));
-    if (authorization)
-    {
-        gchar *path;
-
-        path = get_authorization_path (manager);
-        session_set_authorization (session, authorization, path);
-        g_free (path);
-    }
 
     if (is_greeter)
     {
@@ -152,11 +144,28 @@
         filename = g_strdup_printf ("%s-greeter.log", xserver_get_address (display_get_xserver (display)));
         log_filename = g_build_filename (manager->priv->log_dir, filename, NULL);
         g_free (filename);
+        if (authorization)
+        {
+            gchar *path;
+
+            path = get_authorization_path (manager);
+            session_set_authorization (session, authorization, path);
+            g_free (path);
+        }
     }
     else
     {
         // FIXME: Copy old error file
         log_filename = g_build_filename (user_get_home_directory (session_get_user (session)), ".xsession-errors", NULL);
+        if (authorization)
+        {
+            gchar *path;
+
+            path = g_build_filename(user_get_home_directory (session_get_user (session)), ".Xauthority", NULL);
+            session_set_authorization (session, authorization, path);
+            g_free (path);
+        }
+
     }
 
     if (log_filename)
