Index: lightdm-0.3.0/src/display-manager.c
===================================================================
--- lightdm-0.3.0.orig/src/display-manager.c	2011-04-16 10:16:26.000000000 +0200
+++ lightdm-0.3.0/src/display-manager.c	2011-04-18 15:23:49.470536842 +0200
@@ -144,14 +144,6 @@
     }
 
     authorization = xserver_get_authorization (display_get_xserver (display));
-    if (authorization)
-    {
-        gchar *path;
-
-        path = get_authorization_path (manager);
-        session_set_authorization (session, authorization, path);
-        g_free (path);
-    }
 
     if (is_greeter)
     {
@@ -159,6 +151,14 @@
         filename = g_strdup_printf ("%s-greeter.log", xserver_get_address (display_get_xserver (display)));
         log_filename = g_build_filename (manager->priv->log_dir, filename, NULL);
         g_free (filename);
+        if (authorization)
+        {
+            gchar *path;
+
+            path = get_authorization_path (manager);
+            session_set_authorization (session, authorization, path);
+            g_free (path);
+        }
     }
     else
     {
@@ -172,6 +172,14 @@
                 log_filename = g_build_filename (user_info->pw_dir, ".xsession-errors", NULL);
             else
                 g_warning ("Failed to get user info for user '%s'", session_get_username (session));
+            if (authorization)
+            {
+                gchar *path;
+
+                path = g_build_filename(user_info->pw_dir, ".Xauthority", NULL);
+                session_set_authorization (session, authorization, path);
+                g_free (path);
+            }
         }
     }
   
