Author: Yves-Alexis Perez <corsac@debian.org>
Description: don't add PKGLIBEXEC_DIR to the user PATH
Index: lightdm-1.0.3/src/session.c
===================================================================
--- lightdm-1.0.3.orig/src/session.c	2011-10-08 13:29:42.102575779 +0200
+++ lightdm-1.0.3/src/session.c	2011-10-08 13:29:55.526679659 +0200
@@ -217,26 +217,6 @@
     }
 }
 
-/* Insert our own utility directory to PATH
- * This is to provide gdmflexiserver which provides backwards compatibility
- * with GDM.
- * Must be done after set_env_from_authentication because PAM sets PATH.
- * This can be removed when this is no longer required.
- */
-static void
-insert_utility_path (Session *session)
-{
-    const gchar *orig_path;
-
-    orig_path = session_get_env (session, "PATH");
-    if (orig_path)
-    {
-        gchar *path = g_strdup_printf ("%s:%s", PKGLIBEXEC_DIR, orig_path);
-        session_set_env (session, "PATH", path);
-        g_free (path);
-    }
-}
-
 gboolean
 session_start (Session *session)
 {
@@ -436,7 +416,6 @@
     pam_session_setup (session->priv->authentication);
     set_env_from_authentication (session, session->priv->authentication);
     set_locale (session);
-    insert_utility_path (session);
 
     PROCESS_CLASS (session_parent_class)->run (process);
 }
