diff --git a/src/session.c b/src/session.c
index 4372195..52af4e8 100644
--- a/src/session.c
+++ b/src/session.c
@@ -207,19 +207,6 @@ session_start (Session *session)
     process_set_env (PROCESS (session), "SHELL", user_get_shell (user));
     set_env_from_authentication (session, session->priv->authentication);
 
-    /* Insert our own utility directory to PATH
-     * This is to provide gdmflexiserver which provides backwards compatibility with GDM.
-     * Must be done after set_env_from_authentication because that often sets PATH.
-     * This can be removed when this is no longer required.
-     */
-    orig_path = process_get_env (PROCESS (session), "PATH");
-    if (orig_path)
-    {
-        gchar *path = g_strdup_printf ("%s:%s", PKGLIBEXEC_DIR, orig_path);
-        process_set_env (PROCESS (session), "PATH", path);
-        g_free (path);
-    }
-
     if (session->priv->cookie)
         process_set_env (PROCESS (session), "XDG_SESSION_COOKIE", session->priv->cookie);
 
