Index: lightdm-0.9.5/src/session.c
===================================================================
--- lightdm-0.9.5.orig/src/session.c	2011-09-07 09:21:36.280133903 +0200
+++ lightdm-0.9.5/src/session.c	2011-09-07 09:23:14.224738886 +0200
@@ -209,19 +209,6 @@
     process_set_env (PROCESS (session), "SHELL", user_get_shell (user));
     set_env_from_authentication (session, session->priv->authentication);
 
-    /* Insert our own utility directory to PATH
-     * This is to provide gdmflexiserver which provides backwards compatibility with GDM.
-     * Must be done after set_env_from_authentication because that often sets PATH.
-     * This can be removed when this is no longer required.
-     */
-    orig_path = process_get_env (PROCESS (session), "PATH");
-    if (orig_path)
-    {
-        gchar *path = g_strdup_printf ("%s:%s", PKGLIBEXEC_DIR, orig_path);
-        process_set_env (PROCESS (session), "PATH", path);
-        g_free (path);
-    }
-
     if (session->priv->cookie)
         process_set_env (PROCESS (session), "XDG_SESSION_COOKIE", session->priv->cookie);
 
