Index: lightdm-1.0.1/utils/Makefile.am
===================================================================
--- lightdm-1.0.1.orig/utils/Makefile.am	2011-09-28 09:11:35.000000000 +0200
+++ lightdm-1.0.1/utils/Makefile.am	2011-10-05 22:15:24.740648080 +0200
@@ -1,5 +1,5 @@
 bin_PROGRAMS = dm-tool
-libexec_PROGRAMS = lightdm-set-defaults
+pkglibexec_PROGRAMS = lightdm-set-defaults
 dist_pkglibexec_SCRIPTS = gdmflexiserver
 
 dm_tool_SOURCES = \
