Index: lightdm-1.1.0/tests/src/Makefile.am
===================================================================
--- lightdm-1.1.0.orig/tests/src/Makefile.am	2012-01-14 13:30:41.415305420 +0100
+++ lightdm-1.1.0/tests/src/Makefile.am	2012-01-14 13:30:50.579375996 +0100
@@ -1,5 +1,4 @@
 noinst_PROGRAMS = test-runner X test-gobject-greeter test-session test-script-hook guest-account initctl plymouth
-noinst_SCRIPTS = lightdm-session
 lib_LTLIBRARIES = libsystem.la
 
 libsystem_la_SOURCES = libsystem.c
